doc/sfp-cmis/cmis.md

```diff
@@ -0,0 +1,308 @@
+
+- Provide an unified common SFP parser for the QSFPDD transceivers
+- Enhance the **pmon#xcvrd** for the QSFPDD application initialization sequence
+- Enhance the **pmon#xcvrd** for the QSFPDD diagnostics loopback controls
+
+## 1.1 Functional Requirements
+
+1. Ability to parse the advertised applications from the transceivers
+2. Ability to post the advertised applications to the STATE_DB
+3. Ability to decode the current module state from the transceivers
+4. Ability to post the current module state to the STATE_DB
+5. Ability to decode the error code of the application initialization sequence from the transceivers
+6. Ability to post the error code of the application initialization sequence
+7. Ability to process the application initialization sequence in multi-thread/process model.
+8. Ability to enable / disable the transceiver loopback controls
+
+**Note:**  
+The duration of the CMIS application initialization sequence greatly differs from transceivers
+to transceivers, while some take 3 seconds for activating the 4x100G mode, some require 15 seconds.
+Hence it's mandatory to support multi-threaded/processed models.
+
+## 1.2 Configuration and Management Requirements
+
+1. Show commands to display the QSFPDD EEPROM and DOM information, as described in **Show Commands** section below.
+
+## 1.3 Warm Boot Requirements
+
+Functionality should continue to work across warm boot.
+- To support planned system warm boot.
+- To support pmon docker restart.
+
+# Functionality
+
+## 2.1 Target Deployment Use Cases
+The QSFPDD CMIS support is used to perform application initialization as per the operational
+port breakout mode, otherwise the links will be down if the modules is with CMIS v4/v5 firmwares.
+
+### 2.1.1 CMIS loopback Use Cases
+The loopback controls at the transceiver layer can quickly be tested
+by a remote user to triage either system-side or line-side issues and help isolate failing hardware.
+
+## 2.2 Functional Description
+
+- The **pmon#xcvrd** will detect the module type of the attched transceivers and post the
+information to the **STATE_DB**
+- If the transceiver is a QSFPDD CMIS module, the **pmon#xcvrd** will activate the desired
+application mode as per the current port mode.
+- If the transceiver is a QSFPDD CMIS module, the **show interfaces transceiver eeprom**
+should display the advertised applications of the transceiver.
+- If the transceiver is a QSFPDD CMIS module, the users will be able to enable or disable the
+loopbacks at the transceiver layer to triage link issues.
+- If the transceiver is a QSFPDD CMIS module, the users will be able to check the state of
+the application initialization.
+
+# Design
+
+## 3.1 Overview
+
+![](images/002.png)
+
+### 3.1.1 sonic-platform-daemons
+
+#### 3.1.1.1 sonic-xcvrd (modified)
+
+The transceiver daemon will be enhanced as below:
+
+- Post the QSFP-DD infomration to STATE_DB
+- Add a new CMIS task to handle the multiple CMIS initializations in parallel
```

Comment:
Would you please elaborate the flow of the new task?

Reply:
Please check the following figure in the latest commit, and see if it works, thank you
https://github.com/Azure/SONiC/blob/6eedc2a119f5b5e79fc30da8d2a4747e16c7b603/doc/sfp-cmis/images/001.png